### PR TITLE
UIU-1954: Patron blocks: Allow for override when Renewing

### PR DESCRIPTION
**Pull request by Dmitriy-Litvinenko**
# Description
- Patron blocks: Allow for override of Renewing when logged in user has credentials

# Stories
https://issues.folio.org/browse/UIU-1954

# Screenshots
Implemented for
image
image
Patron Block Modal
image
Override Modal
image

### CODE REVIEW

**Review by annamelnyk — APPROVED** [review]

**Comment by Dmitriy-Litvinenko** [comment]
@zburke @mkuklis Could you please review?

File: src/components/PatronBlock/PatronBlockModalWithOverrideModal.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          open={overrideModal}
+          additionalInfo={additionalInfo}
+          onSetAdditionalInfo={this.onSetAdditionalInfo}
+          onClose={() => {
+            this.onCloseOverrideModal();
+            onOpenPatronBlockedModal();
+          }}
+          onSave={async () => {
+            this.onCloseOverrideModal();
+            this.onSetAdditionalInfo();
```
Comment (mkuklis) [inline src/components/PatronBlock/PatronBlockModalWithOverrideModal.js:55]: @Dmitriy-Litvinenko is there a reason `additionalInfo` is reset to an empty string here?

**Review by mkuklis — APPROVED** [review]
@Dmitriy-Litvinenko this looks good. I noticed that the `additionalInfo` is being passed via many components. It's probably because the way the modals are currently structured. I wonder if in the future we could refactor it a bit to use react context. 

I left one little question for you.

File: src/components/PatronBlock/PatronBlockModalWithOverrideModal.js (hunk 1 above)
Comment (Dmitriy-Litvinenko) [reply src/components/PatronBlock/PatronBlockModalWithOverrideModal.js:55]: We need clear current message after we use renew  loan(s) by barcode. For case that we don't have possible for renew loan(s) by barcode we need use current message for override renewal by barcode. This was discussed with PO.

**Pull request merged**